pennylane/templates/state_preparations/mottonen.py

```diff
@@ -260,45 +255,34 @@ def MottonenStatePreparation(state_vector, wires):
         "".format(expected_shape, get_shape(state_vector)),
     )
 
-    # check if state_vector is normalized
+    # TODO: delete when tape is new core
     if isinstance(state_vector[0], Variable):
-        state_vector_values = [s.val for s in state_vector]
-        norm = np.sum(np.abs(state_vector_values) ** 2)
-    else:
-        norm = np.sum(np.abs(state_vector) ** 2)
+        state_vector = np.array([s.val for s in state_vector])
+
+    # check if normalized
+    norm = np.sum(np.abs(state_vector) ** 2)
     if not np.isclose(norm, 1.0, atol=1e-3):
         raise ValueError("'state_vector' has to be of length 1.0, got {}".format(norm))
 
     #######################
 
-    # Change ordering of indices, original code was for IBM machines
-    state_vector = np.array(state_vector).reshape([2] * n_wires).T.flatten()[:, np.newaxis]
-    state_vector = sparse.dok_matrix(state_vector)
-
-    a = sparse.dok_matrix(state_vector.shape)
-    omega = sparse.dok_matrix(state_vector.shape)
+    # change ordering of wires, since original code was written for IBM machines
+    wires_reverse = wires[::-1]
```

Comment:
This is a much more elegant alternative to reshuffling the state. Thanks @glassnotes for the inspiration!